perception/multi_object_tracker/src/processor/input_manager.cpp

```diff
@@ -0,0 +1,347 @@
+
+  // Update latency statistics
+  // skip initial messages for the latency statistics
+  if (initial_count_ > SKIP_COUNT) {
+    const double latency = (now - objects_time).seconds();
+    if (initial_count_ < INITIALIZATION_COUNT) {
+      // set higher gain for the initial messages
+      constexpr double initial_gain = 0.5;
+      latency_mean_ = (1.0 - initial_gain) * latency_mean_ + initial_gain * latency;
+    } else {
+      constexpr double gain = 0.05;
+      latency_mean_ = (1.0 - gain) * latency_mean_ + gain * latency;
+      const double latency_delta = latency - latency_mean_;
+      latency_var_ = (1.0 - gain) * latency_var_ + gain * latency_delta * latency_delta;
+    }
+  }
+
+  // Calculate interval, Update interval statistics
+  if (initial_count_ > SKIP_COUNT) {
+    const double interval = (now - latest_message_time_).seconds();
+    if (interval < 0.0) {
+      RCLCPP_WARN(
+        node_.get_logger(),
+        "InputManager::updateTimingStatus %s: Negative interval detected, now: %f, "
+        "latest_message_time_: %f",
+        long_name_.c_str(), now.seconds(), latest_message_time_.seconds());
+    } else if (initial_count_ < INITIALIZATION_COUNT) {
+      // Initialization
+      constexpr double initial_gain = 0.5;
+      interval_mean_ = (1.0 - initial_gain) * interval_mean_ + initial_gain * interval;
+    } else {
+      // The interval is considered regular if it is within 0.5 and 1.5 times the mean interval
+      bool update_statistics = interval > 0.5 * interval_mean_ && interval < 1.5 * interval_mean_;
+      if (update_statistics) {
+        constexpr double gain = 0.05;
+        interval_mean_ = (1.0 - gain) * interval_mean_ + gain * interval;
+        const double interval_delta = interval - interval_mean_;
+        interval_var_ = (1.0 - gain) * interval_var_ + gain * interval_delta * interval_delta;
+      }
+    }
+  }
+
+  // Update time
+  latest_message_time_ = now;
+  constexpr double delay_threshold = 3.0;  // [s]
+  if (std::abs((latest_measurement_time_ - objects_time).seconds()) > delay_threshold) {
+    // Reset the latest measurement time if the time difference is too large
+    latest_measurement_time_ = objects_time;
+    RCLCPP_WARN(
+      node_.get_logger(),
+      "InputManager::updateTimingStatus %s: Resetting the latest measurement time to %f",
+      long_name_.c_str(), objects_time.seconds());
+  } else {
+    // Aware reversed message arrival
+    latest_measurement_time_ =
+      latest_measurement_time_ < objects_time ? objects_time : latest_measurement_time_;
+  }
+
+  // Update the initial count
+  if (initial_count_ < INITIALIZATION_COUNT) {
+    initial_count_++;
+  }
+}
+
+void InputStream::getObjectsOlderThan(
+  const rclcpp::Time & object_latest_time, const rclcpp::Time & object_oldest_time,
+  ObjectsList & objects_list)
+{
+  assert(object_latest_time.nanoseconds() > object_oldest_time.nanoseconds());
```

Comment:
In planning simulator, this assertion always occurs in my environment. I appreciate it if someone check if this assertion is valid.

Reply:
cc @YoshiRi @technolojin @yukkysaito @miursh

Reply:
@wep21 Thanks. 
@technolojin May I ask for confirmation?

Reply:
@wep21 @yukkysaito 
Thank you for your report. I will investigate it.

Reply:
@wep21 
I made a fix. https://github.com/autowarefoundation/autoware.universe/pull/7127
However, I couldn't reproduce the situation with the planning simulator. 

Can you test this modification and let me know if the issue is resolved?